### PR TITLE
Phone number formatting

### PR DESCRIPTION
**Pull request by Wcarey7**
- Phone numbers are now auto formatted, using jQuery, as the user types in the new and edit Customer forms. 
The format is: '(XXX)XXX-XXXX'. Also, formats when the field loses focus.

- In the Customer summary page the phone number displayed is formatted. The format is: (XXX)XXX-XXXX

- Function added to the Telephone model to strip the parenthesis and dashes from the phone number and then assigns it to it's phone_number field.

- The phone numbers that are seeded with the customers into the database no longer include parenthesis, dashes, or periods.

- Validation script for the phone number input fields in the new and edit Customer forms. Checks to make sure its 13 characters, the 10 digit phone number, plus the two parenthesis and the dash, or else it throws a validation error on the form.

- Included the email format validation script from the User registration form into the new and edit Customer forms.

- Some misc code organization/cleanup and whitespace deletion.

### CODE REVIEW

**Comment by jdstone** [comment]
@Wcarey7, I promise I'll look at this tomorrow (Saturday).

File: tests/test_customer.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def test_add_customer(client, auth, app):
         "city": "Lancaster",
         "state": "CA",
         "zip": "93536",
-        "phone_number": "6618108496",
+        "phone_number": "(661)810-8496",
```
Comment (jdstone) [inline tests/test_customer.py:19]: Shouldn't this be in the format of ###-###-####?

Comment (Wcarey7) [reply tests/test_customer.py:19]: All the area codes get formatted with parenthesis.

Comment (jdstone) [reply tests/test_customer.py:19]: This all looks good then.

**Review by jdstone — APPROVED** [review]

**Review by jdstone — APPROVED** [review]

**Pull request merged**